presto-hive/src/main/java/com/facebook/presto/hive/EmptyFileCreator.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.metastore.Partition;
+import com.facebook.presto.hive.metastore.Table;
+import com.facebook.presto.spi.ConnectorSession;
+import org.apache.hadoop.fs.Path;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface EmptyFileCreator
```

Comment:
Unfortunately the methods were called `createEmptyFiles`. However it is not true. It doesn't create empty files. It creates "zero row" files. That's why we need to know all the details about the file format. As with some formats zero rows files are not physically empty.

Let's call this interface `ZeroRowFilesCreator`